template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -961,4 +964,27 @@ public Optional<ApiClusterTemplateConfig> getRoleConfig(String serviceType, Stri
                         .filter(config -> configName.equals(config.getName()))
                         .findAny());
     }
+
+    public List<SafetyValve> getSafetyValves() {
+        List<SafetyValve> safetyValves = new ArrayList<>();
+        for (ApiClusterTemplateService apiClusterTemplateService : cmTemplate.getServices()) {
+            (!CollectionUtils.isEmpty(apiClusterTemplateService.getServiceConfigs()) ?
+                    apiClusterTemplateService.getServiceConfigs() : Collections.<ApiClusterTemplateConfig>emptyList()).stream()
+                    .filter(apiClusterTemplateConfig -> apiClusterTemplateConfig.getName().endsWith(SAFETY_VALVE_NAME_SUFFIX))
+                    .forEach(apiClusterTemplateConfig -> safetyValves.add(new SafetyValve(apiClusterTemplateService.getServiceType(), null,
+                            apiClusterTemplateConfig.getName(), apiClusterTemplateConfig.getValue())));
+
+            if (!CollectionUtils.isEmpty(apiClusterTemplateService.getRoleConfigGroups())) {
+                for (ApiClusterTemplateRoleConfigGroup apiClusterTemplateRoleConfigGroup : apiClusterTemplateService.getRoleConfigGroups()) {
+                    (!CollectionUtils.isEmpty(apiClusterTemplateRoleConfigGroup.getConfigs()) ?
+                            apiClusterTemplateRoleConfigGroup.getConfigs() : Collections.<ApiClusterTemplateConfig>emptyList()).stream()
```

Comment:
Same here, please extract the conditional expression into a local variable.

Reply:
Done. Since we are assigning to local variable so we don't need to give type for empty List.

Reply:
This is thread safe. made it final.